packages/cozy-client/src/Schema.js

```diff
@@ -52,13 +53,18 @@ const normalizeDoctypeSchema = doctypeSchema => {
  */
 class Schema {
   constructor(schemaDefinition = {}, client = null) {
+    this.add(schemaDefinition)
+    this.client = client
+  }
+
+  add(schemaDefinition = {}) {
     const values = mapValues(schemaDefinition, (obj, name) => ({
       name,
       ...normalizeDoctypeSchema(obj)
     }))
-    this.client = client
-    this.byName = keyBy(values, x => x.name)
-    this.byDoctype = keyBy(values, x => x.doctype)
+
+    this.byName = merge(this.byName || {}, keyBy(values, x => x.name))
+    this.byDoctype = merge(this.byDoctype || {}, keyBy(values, x => x.doctype))
```

Comment:
Can you add a check to see if some keys are already existing ? I think we should throw in theses cases.